### ISSUE TITLE
Use hex with checksumming

### ISSUE THREAD

**Issue by hackaugusto** [0xMiden/miden-node#28]
Hex encoding will be used on multiple places, account ids, nullifiers, state hashes, etc. When accepting user input it is useful if the data can be checksummed, one approach is to use case of the letters to perform the checksum, similar to how addresses are validated in Ethereum.

**Comment by bobbinth** [comment]
Agreed! Another thing which we could put into the future `miden-utils` crate.

**Comment by hackaugusto** [comment]
some additional motivation: https://github.com/0xPolygonMiden/miden-node/pull/30#pullrequestreview-1555436588

**Comment by hackaugusto** [comment]
Another trade-off we could make is to use a different encoding. Some additional details https://github.com/0xPolygonMiden/miden-base/pull/194#discussion_r1293989915

**Comment by bobbinth** [comment]
Closing this as in protobuf definitions most of the identifies are encoded as bytes rather than strings.
